api/scpca_portal/models/project.py

```diff
@@ -745,6 +463,166 @@ def get_additional_terms(self):
         ) as additional_terms_file:
             return additional_terms_file.read()
 
+    def get_demux_sample_ids(self) -> Set:
+        """Returns a set of all demuxed sample ids used in the project's multiplexed samples."""
+        demux_sample_ids = set()
+        for multiplexed_sample_dir in sorted(Path(self.input_data_path).rglob("*,*")):
+            multiplexed_sample_dir_demux_ids = multiplexed_sample_dir.name.split(",")
+            demux_sample_ids.update(multiplexed_sample_dir_demux_ids)
+
+        return demux_sample_ids
+
+    def get_multiplexed_libraries_metadata(self):
+        """
+        Loads and collects individual multiplexed libraries from json files,
+        then returns them in a list
+        """
+        multiplexed_libraries_metadata = []
+        for multiplexed_sample_dir in sorted(Path(self.input_data_path).rglob("*,*")):
+            for filename_path in sorted(Path(multiplexed_sample_dir).rglob("*_metadata.json")):
+                with open(filename_path) as multiplexed_json_file:
+                    multiplexed_json = json.load(multiplexed_json_file)
+
+                multiplexed_json["scpca_library_id"] = multiplexed_json.pop("library_id")
+                multiplexed_json["scpca_sample_id"] = multiplexed_json.pop("sample_id")
+
+                multiplexed_libraries_metadata.append(multiplexed_json)
+
+        return multiplexed_libraries_metadata
+
+    def get_multiplexed_remaining_fields(self):
```

Comment:
Any reason not to call these aggregate fields?

Reply:
Good call, that's a better name. Will make the change